clippy_lints/src/ok_if_let.rs

```diff
@@ -41,18 +42,36 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for OkIfLet {
     fn check_expr(&mut self, cx: &LateContext<'a, 'tcx>, expr: &'tcx Expr<'_>) {
         if_chain! { //begin checking variables
             if let ExprKind::Match(ref op, ref body, ref source) = expr.kind; //test if expr is a match
-            if let MatchSource::IfLetDesugar { .. } = *source; //test if it is an If Let
-            if let ExprKind::MethodCall(_, _, ref result_types) = op.kind; //check is expr.ok() has type Result<T,E>.ok()
+            if let MatchSource::IfLetDesugar { contains_else_clause } = *source; //test if it is an If Let
+            if let ExprKind::MethodCall(_, ok_span, ref result_types) = op.kind; //check is expr.ok() has type Result<T,E>.ok()
             if let PatKind::TupleStruct(QPath::Resolved(_, ref x), ref y, _)  = body[0].pat.kind; //get operation
             if method_chain_args(op, &["ok"]).is_some(); //test to see if using ok() methoduse std::marker::Sized;
 
             then {
                 let is_result_type = match_type(cx, cx.tables.expr_ty(&result_types[0]), &paths::RESULT);
-                let some_expr_string = snippet(cx, y[0].span, "");
+                let mut applicability = Applicability::MachineApplicable;
+                let some_expr_string = snippet_with_applicability(cx, y[0].span, "", &mut applicability);
+                let trimmed_ok = snippet_with_applicability(cx, op.span.until(ok_span), "", &mut applicability);
+                let mut sugg = format!(
+                    "if let Ok({}) = {} {}",
+                    some_expr_string,
+                    // FIXME(JohnTitor): this trimming is hacky, probably can improve it
+                    trimmed_ok.trim_matches('.'),
+                    snippet(cx, body[0].span, ".."),
+                );
+                if contains_else_clause {
+                    sugg = format!("{} else {}", sugg, snippet(cx, body[1].span, ".."));
+                }
                 if print::to_string(print::NO_ANN, |s| s.print_path(x, false)) == "Some" && is_result_type {
-                    span_help_and_lint(cx, IF_LET_SOME_RESULT, expr.span,
-                    "Matching on `Some` with `ok()` is redundant",
-                    &format!("Consider matching on `Ok({})` and removing the call to `ok` instead", some_expr_string));
+                    span_lint_and_sugg(
+                        cx,
+                        IF_LET_SOME_RESULT,
+                        expr.span,
```

Comment:
Hey what are your thoughts on reducing the span of this lint to just the `if let Some(y) = x.parse().ok()` portion since there are no changes required within the if/else blocks. I think spanning the entire expression would be noisy if the if/else blocks are huge but maybe I am missing something.

Reply:
Fair enough, reduced span size via 08680cfdc90ea2e03bec5aaa0ee88852e3a41f2a